app-modules/meeting-center/src/Filament/Resources/CalendarEventResource/Pages/ListCalendarEvents.php

```diff
@@ -0,0 +1,133 @@
+{
+    protected static string $resource = CalendarEventResource::class;
+
+    protected static string $view = 'meeting-center::filament.pages.list-events';
+
+    #[Url(as: 'view')]
+    public string $viewType = 'table';
+
+    public function setupCalendarProviderAction(): Action
+    {
+        return Action::make('setupCalendarProviderAction')
+            ->modalHeading('Choose a provider')
+            ->modalDescription('This feature requires synchronization with Google or Outlook. Please select the service you would like
+        to synchronize with below to continue.')
+            ->modalContent(view('meeting-center::filament.components.calendar-setup-modal'))
+            ->modalSubmitAction(false)
+            ->modalCancelAction(Action::make('cancel')->color('gray')->url(Filament::getUrl()))
+            ->modalCloseButton(false)
+            ->closeModalByClickingAway(false);
+    }
+
+    public function selectCalendarAction(): Action
+    {
+        return Action::make('selectCalendarAction')
+            ->modalHeading('Select a Calendar')
+            ->form([
+                Radio::make('provider_id')
+                    ->hiddenLabel()
+                    ->options(function () {
+                        /** @var User $user */
+                        $user = auth()->user();
+
+                        $calendar = $user->calendar;
+
+                        return resolve(CalendarManager::class)
+                            ->driver($calendar->provider_type->value)
+                            ->getCalendars($calendar);
+                    })
+                    ->required(),
+            ])
+            ->modalCloseButton(false)
+            ->closeModalByClickingAway(false)
+            ->modalCancelAction(Action::make('cancel')->color('gray')->url(Filament::getUrl()))
+            ->action(function (array $data): void {
+                /** @var User $user */
+                $user = auth()->user();
+
+                $calendar = $user->calendar;
+                $calendar->provider_id = $data['provider_id'];
+
+                $calendars = resolve(CalendarManager::class)
+                    ->driver($calendar->provider_type->value)
+                    ->getCalendars($calendar);
+
+                $calendar->name = $calendars[$data['provider_id']];
+
+                $calendar->saveQuietly();
+
+                resolve(CalendarManager::class)
+                    ->driver($calendar->provider_type->value)
+                    ->syncEvents($calendar);
+            });
+    }
+
+    public function setViewType(string $viewType): void
+    {
+        $this->viewType = $viewType;
+    }
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->columns([
+                IdColumn::make(),
+                TextColumn::make('title'),
+                TextColumn::make('starts_at'),
+                TextColumn::make('ends_at'),
+            ])
+            ->filters([
+            ])
```

Comment:
Should we create a filter to only show events that are either currently happening OR in the future? That way the list view isn't full of past events.

And then set that filter to be on by default.

Reply:
I added a filter that hides all events before the current day by default.